### ISSUE TITLE
Skip filtered function from other task when replay

### ISSUE THREAD

**Issue by namhyung** [namhyung/uftrace#27]
Currently uftrace replay merges ENTRY and EXIT records if the functions is a leaf or all functions between them are filtered out (and then it'll look like a leaf).  But it only checks functions from same task and didn't merge when there's another record of other task even if it's filtered out (so won't be shown).  This can be easily seen when kernel tracing is enabled.

A Simpler example is the multi-thread test case (t-thread).

```
$ uftrace record tests/t-thread

$ uftrace replay
# DURATION    TID     FUNCTION
            [20053] | main() {
  81.953 us [20053] |   pthread_create();
  53.108 us [20053] |   pthread_create();
 188.065 us [20053] |   pthread_create();
 184.846 us [20053] |   pthread_create();
            [20053] |   pthread_join() {
            [20057] | foo() {
            [20057] |   a() {
            [20057] |     b() {
   5.735 us [20057] |       c();
   7.308 us [20057] |     } /* b */
   8.081 us [20057] |   } /* a */
   9.212 us [20057] | } /* foo */
            [20056] | foo() {
            [20056] |   a() {
            [20056] |     b() {
   0.211 us [20056] |       c();
   1.069 us [20056] |     } /* b */
   1.480 us [20056] |   } /* a */
   2.086 us [20056] | } /* foo */
            [20055] | foo() {
            [20055] |   a() {
            [20055] |     b() {
   0.335 us [20055] |       c();
   1.623 us [20055] |     } /* b */
   2.551 us [20055] |   } /* a */
   3.944 us [20055] | } /* foo */
            [20058] | foo() {
            [20058] |   a() {
            [20058] |     b() {
   0.193 us [20058] |       c();
   1.387 us [20058] |     } /* b */
   1.886 us [20058] |   } /* a */
   2.562 us [20058] | } /* foo */
 779.561 us [20053] |   } /* pthread_join */
   1.136 us [20053] |   pthread_join();
   0.702 us [20053] |   pthread_join();
   0.650 us [20053] |   pthread_join();
   1.309 ms [20053] | } /* main */
```

When I set function filter on "main" function, it only shows the main thread but pthread_join() cannot be merged since there are many functions (not displayed though) from other tasks.

```
$ uftrace replay -F main
# DURATION    TID     FUNCTION
            [20053] | main() {
  81.953 us [20053] |   pthread_create();
  53.108 us [20053] |   pthread_create();
 188.065 us [20053] |   pthread_create();
 184.846 us [20053] |   pthread_create();
            [20053] |   pthread_join() {
 779.561 us [20053] |   } /* pthread_join */
   1.136 us [20053] |   pthread_join();
   0.702 us [20053] |   pthread_join();
   0.650 us [20053] |   pthread_join();
   1.309 ms [20053] | } /* main */
```

With this patchset, it can skip those filtered functions and can merge pthread_join().

```
$ uftrace replay -F main
# DURATION    TID     FUNCTION
            [20053] | main() {
  81.953 us [20053] |   pthread_create();
  53.108 us [20053] |   pthread_create();
 188.065 us [20053] |   pthread_create();
 184.846 us [20053] |   pthread_create();
 779.561 us [20053] |   pthread_join();
   1.136 us [20053] |   pthread_join();
   0.702 us [20053] |   pthread_join();
   0.650 us [20053] |   pthread_join();
   1.309 ms [20053] | } /* main */
```

I pushed the code to the `review/replay-skip` branch.

**Comment by honggyukim** [comment]
Thanks for the fix. I was aware of this problem, didn't report it though. I just tested it and found that the problem is gone now. Thanks!

**Comment by namhyung** [comment]
Merged: 6e4a1c13
